Create features and bundles for MP Fault Tolerance 2.1

Azquelt opened the pull request:
This includes refactoring the existing Fault Tolerance bundles where required to allow them to be extended for Fault Tolerance 2.1, resulting in removing copying between bundles and removing the unused failsafe dependency from fault tolerance 2.0.

Also renamed the Fault Tolerance CDI integration bundles to have a more consistent naming structure.

Fixes #9672 
Fixes #5529
Fixes #5538
Fixes #6182

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpFaultTolerance-2.1/resources/l10n/com.ibm.websphere.appserver.mpFaultTolerance-2.0.properties:
```diff
@@ -0,0 +1,17 @@
+###############################################################################
```
File is named incorrectly

Azquelt replied:
Done

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpFaultTolerance-2.1/resources/l10n/com.ibm.websphere.appserver.mpFaultTolerance-2.0.properties:
```diff
@@ -0,0 +1,17 @@
+###############################################################################
+# Copyright (c) 2018 IBM Corporation and others.
```
Wrong date

Azquelt replied:
Done

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpFaultTolerance-2.1/com.ibm.websphere.appserver.mpFaultTolerance-2.0.feature:
```diff
@@ -0,0 +1,24 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
```
File is named incorrectly

Azquelt replied:
Done

tevans78 commented on dev/com.ibm.websphere.org.eclipse.microprofile.faulttolerance.2.1/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/bin/
```
This .gitignore may not be needed as these paths may be covered at the top level

Azquelt replied:
You are correct, it is not needed.

tevans78 commented on dev/com.ibm.websphere.org.eclipse.microprofile.faulttolerance.2.1/bnd.bnd:
```diff
@@ -0,0 +1,38 @@
+Bundle-SymbolicName: com.ibm.websphere.org.eclipse.microprofile.faulttolerance.2.1; singleton:=true
+
+Export-Package: \
+  org.eclipse.microprofile.faulttolerance;version=1.1, \
```
I think this should be at least version 1.2

Azquelt replied:
Done, though should be reviewed when we update to the released version

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services/bnd.bnd:
```diff
@@ -0,0 +1,35 @@
+#*******************************************************************************
+# Copyright (c) 2017,2018 IBM Corporation and others.
```
date

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services/src/com/ibm/ws/micrprofile/faulttolerance21/cdi/services/AnnotationConfigFactoryComponent.java:
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
date

Azquelt replied:
Did this and the other files in the same package.

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1/bnd.bnd:
```diff
@@ -0,0 +1,49 @@
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.cdi.2.1
+Bundle-SymbolicName: com.ibm.ws.microprofile.faulttolerance.cdi.2.1
+Bundle-Description: MicroProfile Fault Tolerance CDI 2.0 Integration, version ${bVersion}
```
wrong version in the description

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1/bnd.bnd:
```diff
@@ -0,0 +1,49 @@
+#*******************************************************************************
```
Please change the naming of these projects so that it follows similar conventions to MP Config.
so this one would be `com.ibm.ws.microprofile.faulttolerance.2.1.cdi`. This makes it clear that it is a FT 2.1 project and not an integration with CDI 2.1

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/package-info.java:
```diff
@@ -11,7 +11,7 @@
 /**
  * @version 2.0
```
Shouldn't this version match the one below?

Azquelt replied:
Err, yeah...

Azquelt replied:
Yes it should.

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/bin/
```
Not needed

Azquelt replied:
Done

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services/bnd.bnd:
```diff
@@ -0,0 +1,35 @@
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services
+Bundle-SymbolicName: com.ibm.ws.microprofile.faulttolerance.cdi.2.1.services
+Bundle-Description: MicroProfile Fault Tolerance CDI Integration, version ${bVersion} services
```
Should the bundle description give the FT version?

Azquelt replied:
done

Azquelt commented:
#build

Azquelt commented:
@tevans78 I think that's everything addressed.

Azquelt commented:
#build

Azquelt commented:
Corrected the ft-2.1 feature file so it doesn't reference a non-existent bundle.

#build

Azquelt commented:
Rebased on master, added parallel activation to the new 2.1 feature to match #9558, fixed one place where the bundle names had not been updated, fix the 2.1 bundles pulling in the 1.1 API and added `@TraceOptions` to new package.

#build

Azquelt commented:
#build

Azquelt commented:
#run-libby-bot
